components/eamxx/src/physics/dp/dp_functions_f90.hpp

```diff
@@ -84,15 +84,59 @@ struct AdvanceIopSubsidenceData : public PhysicsTestData {
 
 struct IopSetinitialData : public PhysicsTestData {
   // Inputs
-  Int plev, nelemd;
+  Int plev, pcnst, nelemd, np, nstep;
+
+  bool use_replay, dynproc, have_t, have_q, have_ps, have_u, have_v, have_numliq, have_cldliq, have_numice, have_cldice, scm_zero_non_iop_tracers, is_first_restart_step;
+
+  Real psobs;
+
+  Real* qmin, *uobs, *vobs, *numliqobs, *numiceobs, *cldliqobs, *cldiceobs, *dx_short;
 
   // Inputs/Outputs
-  element_t *elem;
+  tracer_t tracers;
+  element_t elem;
+
+  Real dyn_dx_size;
+
+  Real* tobs, *qobs;
+
+  IopSetinitialData(
+    Int plev_, Int pcnst_, Int nelemd_, Int np_, Int nstep_, Real psobs_,
+    bool use_replay_, bool dynproc_, bool have_t_, bool have_q_, bool have_ps_, bool have_u_, bool have_v_, bool have_numliq_, bool have_cldliq_, bool have_numice_, bool have_cldice_, bool scm_zero_non_iop_tracers_, bool is_first_restart_step_) :
+    PhysicsTestData(
+      {{nelemd_}, {pcnst_}, {plev_}},
+      {
+        {&dx_short},
+        {&qmin},
+        {&uobs, &vobs, &numliqobs, &numiceobs, &cldliqobs, &cldiceobs, &tobs, &qobs}
+      }),
+    plev(plev_), pcnst(pcnst_), nelemd(nelemd_), np(np_), nstep(nstep_), psobs(psobs_),
+    use_replay(use_replay_), dynproc(dynproc_), have_t(have_t_), have_q(have_q_), have_ps(have_ps_), have_u(have_u_), have_v(have_v_), have_numliq(have_numliq_), have_cldliq(have_cldliq_), have_numice(have_numice_), have_cldice(have_cldice_), scm_zero_non_iop_tracers(scm_zero_non_iop_tracers_), is_first_restart_step(is_first_restart_step_) { }
+
+  PTD_STD_DEF(IopSetinitialData, 19, plev, pcnst, nelemd, np, nstep, psobs, use_replay, dynproc, have_t, have_q, have_ps, have_u, have_v, have_numliq, have_cldliq, have_numice, have_cldice, scm_zero_non_iop_tracers, is_first_restart_step);
+
+  void init()
+  {
+    tracers.init(nelemd, 10); // what is num tracers?
```

Comment:
Guessed at few things here. I don't know where I'd get num tracers from. I also didn't know what to put for the Element scale factor.